tests/experimental/datasources/test_pandas_datasource.py

```diff
@@ -1,20 +1,41 @@
+from __future__ import annotations
+
+import inspect
+import logging
 import pathlib
 from dataclasses import dataclass
-from typing import List
+from pprint import pformat as pf
+from typing import TYPE_CHECKING, Any
 
+import pandas as pd
+import pydantic
 import pytest
+from pytest import MonkeyPatch, param
 
 import great_expectations.exceptions as ge_exceptions
-from great_expectations.alias_types import PathStr
+import great_expectations.execution_engine.pandas_execution_engine
 from great_expectations.data_context.util import file_relative_path
-from great_expectations.experimental.datasources.interfaces import (
-    BatchSortersDefinition,
-)
 from great_expectations.experimental.datasources.pandas_datasource import (
     CSVAsset,
     PandasDatasource,
 )
 
+if TYPE_CHECKING:
+    from great_expectations.alias_types import PathStr
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSortersDefinition,
+    )
+
+logger = logging.getLogger(__file__)
+
+# TODO: use this to parametrize `test_data_asset_defined_for_io_read_method`
+PANDAS_IO_METHODS: list[str] = [
```

Comment:
Is this used anywhere, can we delete it?

Reply:
Will do